### ISSUE TITLE
Half - Life 2 [Half-Life 2] [45410091]

### ISSUE THREAD

**Issue by hotzenplotz3000** [Cxbx-Reloaded/game-compatibility#538]
[Half-Life 2](https://en.wikipedia.org/wiki/Half-Life_2)

### Description

This game uses two XBEs.
Default.xbe starts and displays some splash screens and videos, but the videos only show a black screen.
Then it starts the hl2_xbox.xbe, which uses LTCG.

### Dumps 

[default.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627938/default.txt)
[hl2_xbox.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627939/hl2_xbox.txt)
[KrnlDebug_default_xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627941/KrnlDebug_default_xbe.txt)
[hlecache_default_xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627942/hlecache_default_xbe.txt)

### Status
**Previous behavior:** 
interactive in #417 

**Current behavior:**
main xbe uses LTCG

### Screenshots 
![2018-01-13 02_16_48-cxbx-reloaded 5ce5da70 jan 12 2018 _ emulating half - life 2 - fps_ 59 90 ms_](https://user-images.githubusercontent.com/30478365/34901405-2c9d7822-f80a-11e7-8990-74f0afb5417c.png)

### Additional Information

tested with build 5ce5da70

**Comment by CakeLancelot** [comment]
This one is formatted much better, thanks!

**Comment by haradons** [comment]
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2019758/KrnlDebug.txt)
Build: 46c079cf
Gets to menu, however after it loads console displays:
[0x0070] MAIN: Received Breakpoint Exception (int 3)

 EIP := 0x00596301(=unknown+0x0)
 EFL := 0x00000212
 EAX := 0x0B84F601 EBX := 0x00000000 ECX := 0x0B84F62C EDX := 0x0B84F679
 ESI := 0x00010000 EDI := 0x009AA91E ESP := 0x0B84F664 EBP := 0xFFFFFFFF
 CR2 := 0x00000000

then crashes
![image](https://user-images.githubusercontent.com/13687549/40272140-bbcdf52e-5b5d-11e8-86f9-3636a4c4300b.png)
![image](https://user-images.githubusercontent.com/13687549/40272141-c386cb6a-5b5d-11e8-9e44-6b3e09e2a73c.png)

**Comment by JusicP** [comment]
I replaced 10 bytes at 0x00596301 with NOP to avoid _debugbreak calling and game started loading further(game reads cfg and resource files from zip0_xbox.xzp). But then a lot of EmuX86_Mem_Read Failed (0xFE820010, 4) messages appears in console.

**Comment by JusicP** [comment]
Half-Life 2 goes ingame with black screen.
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/3830243/KrnlDebug.txt)

**Comment by JusicP** [comment]
The game crashes after FatalError calling in this function. 
![image](https://user-images.githubusercontent.com/25232950/68886346-bb8a3900-071f-11ea-8b07-81808a8de09c.png)

Here is the place where function is called:
![image](https://user-images.githubusercontent.com/25232950/68886373-c80e9180-071f-11ea-826a-f55cd30f5d66.png)

**Comment by JusicP** [comment]
Here are the results with LLE GPU enabled:
![image](https://user-images.githubusercontent.com/25232950/68942989-998dc680-07b2-11ea-9381-4df9b7533605.png)
![image](https://user-images.githubusercontent.com/25232950/68943066-cf32af80-07b2-11ea-877f-5dcc92a9d898.png)

**Comment by CrazyCSIW6** [comment]
> Here are the results with LLE GPU enabled:
> ![image](https://user-images.githubusercontent.com/25232950/68942989-998dc680-07b2-11ea-9381-4df9b7533605.png)
> ![image](https://user-images.githubusercontent.com/25232950/68943066-cf32af80-07b2-11ea-877f-5dcc92a9d898.png)

MY EYES

**Comment by T02my** [comment]
![image](https://user-images.githubusercontent.com/26767993/79047618-6f84e700-7c20-11ea-9072-427fb9e40a43.png)

**Comment by CrazyCSIW6** [comment]
I don't know what looks worse. This, or the game on an actual original Xbox.
At least it loads and plays with LLE GPU enabled. That's a plus.

**Comment by T02my** [comment]
The game runs with HLE. On actual original Xbox the game does not look that bad, it's not masterpiece like Doom 3 on Xbox, but still.

**Comment by PatrickvL** [comment]
Shall we discuss the weather was well here?..... The point is, none of these remarks add anything of value to the issue. Please, only comment with facts and perhaps some evidence to back that up(like screenshots), but refrain from personal preferences and opinions.

**Comment by bryce13950** [comment]
The most recent build seems to have some regression. I only was given a black screen.

[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/4777191/KrnlDebug.txt)

**Comment by FerPozzetti** [comment]
> The most recent build seems to have some regression. I only was given a black screen.
> 
> [KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/4777191/KrnlDebug.txt)

Inside the debug file it says:

"[0x1418] WARN : INIT    Invalid xbe signature. Homebrew, tampered or pirated xbe?"

Is that normal in this game? Or is your copy modified?

**Comment by gandalfthewhite19890404** [comment]
@Margen67 
Why do you approving regressions without confirmation from other users?
Testers - when you noticing any regression - please test much more than once!
Actual state - ingame, game may crash after some time of gameplay.
![image](https://user-images.githubusercontent.com/24764335/84722897-7835c880-af95-11ea-8fcb-95da3ac178a0.png)

**Comment by bryce13950** [comment]
> @Margen67
> Why do you approving regressions without confirmation from other users?
> Testers - when you noticing any regression - please test much more than once!
> Actual state - ingame, game may crash after some time of gameplay. With all cores hack game works much faster - up to 20 fps, but I have got hang when Gordon encountered Barney, without hack - you may not get until that moment at all.
> ![image](https://user-images.githubusercontent.com/24764335/84722897-7835c880-af95-11ea-8fcb-95da3ac178a0.png)

Do you have any hacks enabled? My regression came after I freshly restored my default options, and tested with vanilla settings.

**Comment by gandalfthewhite19890404** [comment]
> Do you have any hacks enabled? My regression came after I freshly restored my default options, and tested with vanilla settings.

No, hack is need just for speedup (on screenshot you may see fps without all cores hack), game boots without any special settings.

**Comment by bryce13950** [comment]
I do not know what the issue is with this one, but I continue to only receive a black screen even on a brand new computer that I received since the last time I tested. I do not have any abnormal issues on about 50 other titles I have been testing.

**Comment by mnadareski** [comment]
Using build df22c8b:
- The opening cutscene displays an all-black screen, though the G-Man audio still plays
- After the level loads, models and environment appear to be loading correctly, though with potentially odd shadows (see the right side of the train car)
![image](https://user-images.githubusercontent.com/4381583/119388107-b431dd00-bc7e-11eb-93ab-572efdbdd10e.png)
- Character and object shadows are all blocks
![image](https://user-images.githubusercontent.com/4381583/119388278-e9d6c600-bc7e-11eb-82c5-e045ad9519d5.png)
- Audio starts to layer after a while, so most of the sounds and dialog compound and play at full volume
- Once you are diverted from going to the train to Nova Prospekt, the game froze unexpectedly when following down the corridor

Notable messages:
```
[0x5B0C] INFO : VTXSH   Please report that Half Life 2 - Retail Build shows the following message:
LOG_TEST_CASE: FVF without position
In XboxVertexShaderFromFVF (D:\a\Cxbx-Reloaded\Cxbx-Reloaded\src\core\hle\D3D8\XbVertexShader.cpp line 130)
[0x5B0C] INFO : VTXSH   Please report that Half Life 2 - Retail Build shows the following message:
LOG_TEST_CASE: Unassigned Xbox vertex shader!
In GetXboxVertexShader (D:\a\Cxbx-Reloaded\Cxbx-Reloaded\src\core\hle\D3D8\XbVertexShader.cpp line 268)
[0x5B0C] INFO : VTXSH   Please report that Half Life 2 - Retail Build shows the following message:
LOG_TEST_CASE: Xbox should always have a VertexShader set (even for FVF's)
In GetXboxVertexAttributeFormat (D:\a\Cxbx-Reloaded\Cxbx-Reloaded\src\core\hle\D3D8\XbVertexShader.cpp line 356)
[0x5B0C] INFO : D3D8    Please report that Half Life 2 - Retail Build shows the following message:
LOG_TEST_CASE: Unknown texture stage!
In GetHostResourceKey (D:\a\Cxbx-Reloaded\Cxbx-Reloaded\src\core\hle\D3D8\Direct3D9\Direct3D9.cpp line 978)
```

Kernel Debug: [KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/6534148/KrnlDebug.txt)
